csharp/test/Apache.Arrow.Tests/BooleanArrayTests.cs

```diff
@@ -124,12 +126,14 @@ public class Swap
                 public void SwapsExpectedBits()
                 {
                     var array = new BooleanArray.Builder()
-                        .Resize(8)
+                        .AppendRange(Enumerable.Repeat(false, 8))
```

Comment:
Changed because increasing the size of a builder via `Resize()` creates new elements that have their validity set to `false`.

The [existing docs don't state](https://arrow.apache.org/docs/cpp/api/builder.html#_CPPv4N5arrow12ArrayBuilder6ResizeE7int64_t) what happens to validity upon resizing, so I assume this is appropriate behaviour.